CIAOS-Rust/server/src/database.rs

```diff
@@ -0,0 +1,105 @@
+//database.rs
+
+use std::sync::Mutex;
+use rusqlite::{params, Connection, Result as SqliteResult};
+use std::sync::Arc;
+use log::warn;
+use actix_web::Error;
+
+
+pub struct Database {
+    db_conn: Arc<Mutex<Connection>>,
+}
+
+impl Database {
+
+    pub fn new(user: &str) -> Result<Self, actix_web::Error> {
+            let conn = Connection::open(format!("{}.sqlite", user)).expect("Failed to open the database");
+            conn.execute(
+                "CREATE TABLE IF NOT EXISTS metadata (
```

Comment:
where is the user column created? How will we know if a particular key exists or not without identifying by users?